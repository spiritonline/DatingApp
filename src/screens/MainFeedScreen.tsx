import { SafeAreaProvider } from 'react-native-safe-area-context';
import { MainTabNavigator } from '../navigation/MainTabNavigator';
import { Provider } from 'react-redux';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { store } from '../store';

// Create a client for React Query
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: 5 * 60 * 1000, // 5 minutes
      retry: 1,
      refetchOnWindowFocus: false,
    },
  },
});

/**
 * Main Feed Screen - entry point for the authenticated app experience
 * This wraps the tab navigation in the required providers for Redux and React Query
 */
export default function MainFeedScreen() {
  return (
    <Provider store={store}>
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider>
          <MainTabNavigator />
        </SafeAreaProvider>
      </QueryClientProvider>
    </Provider>
  );
}
