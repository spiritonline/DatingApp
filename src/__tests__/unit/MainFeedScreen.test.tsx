import React from 'react';
import { render } from '@testing-library/react-native';
import MainFeedScreen from '../../screens/MainFeedScreen';
import { NavigationContainer } from '@react-navigation/native';

describe('MainFeedScreen', () => {
  it('renders correctly with title and subtitle', () => {
    const { getByText } = render(
      // MainFeedScreen is simple and doesn't use navigation hooks directly,
      // but wrapping in NavigationContainer is good practice for screens.
      <NavigationContainer>
        <MainFeedScreen />
      </NavigationContainer>
    );

    expect(getByText('Main Feed')).toBeTruthy();
    expect(getByText('Authentication successful!')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <MainFeedScreen />
      </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
