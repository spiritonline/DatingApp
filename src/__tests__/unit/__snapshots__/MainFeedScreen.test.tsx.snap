// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainFeedScreen matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      Main Feed
    </Text>
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 16,
        }
      }
    >
      Authentication successful!
    </Text>
  </View>
</RNCSafeAreaView>
`;
